### PR TITLE
Users/sarojkpr/broadcast service  widget isolation

### PR DESCRIPTION
**Pull request by sarojkpr**
Updating the Broadcast Service for multi widget/multi-tab scenario for listener isolation. This is needed to avoid widget2 also listening to SDK methods from same Broadcast channel.

### CODE REVIEW

File: chat-components/src/services/BroadcastService.ts [hunk 1]
```diff
@@ -2,16 +2,40 @@ import { Subject } from "rxjs";
 import { ICustomEvent } from "../interfaces/ICustomEvent";
 import { filter } from "rxjs/operators";
 import { BroadcastChannel } from "broadcast-channel";
-import { BROADCAST_CHANNEL_NAME } from "../common/Constants";
```
Comment (charliewang95) [inline chat-components/src/services/BroadcastService.ts:5]: If not used anymore, can we remove from Constants?

**Review by charliewang95 — APPROVED** [review]

File: chat-components/src/services/BroadcastService.ts [hunk 2]
```diff
@@ -2,16 +2,40 @@ import { Subject } from "rxjs";
...
+const broadcastServiceSubList: Record<string, any> = {};
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-subChannel.onmessage = (message: any) => {
-    newMessage.next(message);
+let pubChannel: any;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let subChannel: any;
+
+export const BroadcastServiceInitialize = (channelName: string) => {
```
Comment (charliewang95) [inline chat-components/src/services/BroadcastService.ts:18]: Please check if all lcw events are still being raised from script layer after this change

**Review by charliewang95 — APPROVED** [review]

File: chat-components/src/services/BroadcastService.ts (hunk 1 above)
Comment (sarojkpr) [reply chat-components/src/services/BroadcastService.ts:5]: removed this

File: chat-components/src/services/BroadcastService.ts (hunk 2 above)
Comment (sarojkpr) [reply chat-components/src/services/BroadcastService.ts:18]: okay, will validate them before raising widget PR

**Pull request merged**